compiler/src/main/java/org/neo4j/ogm/compiler/MultiStatementCypherCompiler.java

```diff
@@ -100,10 +102,8 @@ public void unmap(NodeBuilder nodeBuilder) {
 		List<Statement> statements = new ArrayList<>(newNodesByLabels.size());
 		for (Set<Node> nodeModels : newNodesByLabels.values()) {
 			NewNodeEmitter newNodeEmitter = new NewNodeEmitter(nodeModels);
-			final Map<String, Object> parameters = new HashMap<>();
-			final StringBuilder query = new StringBuilder();
-			newNodeEmitter.emit(query, parameters);
-			statements.add(statementFactory.statement(query.toString(), parameters));
+			Pair<String, Map<String, Object>> emittedCypher =  newNodeEmitter.emit();
+			statements.add(statementFactory.statement(emittedCypher.first, emittedCypher.second));
```

Comment:
if `emit()` used a Pair internally (for the sake of argument) it could then use `statementFactory.statement(pair.first, pair.second)` and emit an actual Statement:

`statements.add(newNodeEmitter.emit())`

We could then make the xxxEmitter classes more like proper statment builders, e.g.: 

`statements.add(newNodeStatements().build())`